### PR TITLE
Impalav2 integration

### PR DESCRIPTION
**Pull request by radroof22**
Implementation of Impala in StableBaselines3 inspired from the [coinrun implementation](https://github.com/AurelianTactics/contra_3_rl/blob/master/ppo2_contra_baselines_impala_lvl1.py). [Blog](https://medium.com/aureliantactics/custom-models-with-baselines-impala-cnn-cnns-with-features-and-contra-3-hard-mode-811dbdf2dff9)

The problem with the above is that it is using Tensorflow. This means we have to port it over to PyTorch.

I tried doing this quickly with ChatGPT but it's ignoring the special dropout protocol they use for coinrun. Rather than having the random coin run, to ensure deterministic behavior across checkpoints they save a [fixed mask for dropout](https://github.com/openai/coinrun/blob/master/coinrun/policies.py#L19-L38).

### CODE REVIEW

File: training-pipeline/impala.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import torch.nn.functional as F
+import numpy as np
+from stable_baselines3.common.policies import ActorCriticPolicy
+from gym.spaces import Box
+
+
+class Config:
+    USE_BATCH_NORM = 1
+    DROPOUT = 0
+    ARCHITECTURE = 'impala'
```
Comment (radroof22) [inline training-pipeline/impala.py]: convert this to a .env file reading in values

File: training-pipeline/impala.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        super().__init__()
+        self.conv1 = nn.Conv2d(in_channels, in_channels, 3, padding=1)
+        self.conv2 = nn.Conv2d(in_channels, in_channels, 3, padding=1)
+        self.use_batch_norm = use_batch_norm
+        
+        if use_batch_norm:
+            self.bn1 = nn.BatchNorm2d(in_channels)
+            self.bn2 = nn.BatchNorm2d(in_channels)
+            
+        self.dropout = nn.Dropout2d(p=Config.DROPOUT) if Config.DROPOUT > 0 else None
```
Comment (radroof22) [inline training-pipeline/impala.py]: Dropout still not working with the [fixed dropout mask per batch](https://github.com/openai/coinrun/blob/master/coinrun/policies.py#L9C1-L38C19).

Comment (radroof22) [reply training-pipeline/impala.py]: Based on conversation with @abhay-sheshadri, keep the nn.Dropout and just ensure we can converge after some time. Otherwise will have to choose a different environment.
